core/templates/components/forms/custom-forms-directives/thumbnail-uploader.component.ts

```diff
@@ -34,37 +34,37 @@ import { EditThumbnailModalComponent } from './edit-thumbnail-modal.component';
   templateUrl: './thumbnail-uploader.component.html'
 })
 export class ThumbnailUploaderComponent implements OnInit, OnChanges {
-  @Input() disabled: boolean;
-  @Input() useLocalStorage: boolean;
-  @Input() allowedBgColors: string[];
-  @Input() aspectRatio: string;
-  @Input() bgColor: string;
-  @Input() filename: string;
-  @Input() previewDescription: string;
-  @Input() previewDescriptionBgColor: string;
-  @Input() previewFooter: string;
-  @Input() previewTitle: string;
-  @Output() updateBgColor: EventEmitter<string> = new EventEmitter() ;
+  @Input() disabled!: boolean;
+  @Input() useLocalStorage!: boolean;
+  @Input() allowedBgColors!: string[];
+  @Input() aspectRatio!: string;
+  @Input() bgColor!: string;
+  @Input() filename!: string;
+  @Input() previewDescription!: string;
+  @Input() previewDescriptionBgColor!: string;
+  @Input() previewFooter!: string;
+  @Input() previewTitle!: string;
+  @Output() updateBgColor: EventEmitter<string> = new EventEmitter();
   @Output() updateFilename: EventEmitter<string> = new EventEmitter();
   @Output() imageSave: EventEmitter<void> = new EventEmitter();
-  openInUploadMode: boolean;
-  tempBgColor: string;
-  tempImageName: string;
-  uploadedImage: string;
-  uploadedImageMimeType: string;
-  dimensions: { height: number; width: number };
-  resampledFile: Blob;
-  newThumbnailDataUrl: string;
-  localStorageBgcolor: string;
-  imageUploadUrlTemplate: string;
+  openInUploadMode: boolean | undefined;
+  tempBgColor: string | undefined;
+  tempImageName: string | undefined;
+  uploadedImage: string | undefined;
+  uploadedImageMimeType: string | undefined;
+  dimensions: { height: number; width: number } | undefined;
+  resampledFile: Blob | undefined;
+  newThumbnailDataUrl: string | undefined;
+  localStorageBgcolor: string | undefined;
+  imageUploadUrlTemplate: string | undefined;
   hidePlaceholder = true;
   placeholderImageUrl = (
     this.urlInterpolationService.getStaticImageUrl(
       '/icons/story-image-icon.png'));
-  editableThumbnailDataUrl: string;
-  transformedData: string;
-  parsedResponse;
-  encodedImageURI: string;
+  editableThumbnailDataUrl: string | undefined;
+  transformedData: string | undefined;
+  parsedResponse: unknown;
```

Comment:
Why unknown?

Reply:
I could not figure out what type it is

Reply:
I think you will need to focus on this more since we don't want to allow `unknowns` that much in our codebase.